exporter/clickhouseexporter/internal/metrics_model.go

```diff
@@ -0,0 +1,467 @@
+%s
+PARTITION BY toDate(TimeUnix)
+ORDER BY (MetricName, Attributes, toUnixTimestamp64Nano(TimeUnix))
+SETTINGS index_granularity=8192, ttl_only_drop_parts = 1;
+`
+)
+
+const (
+	// language=ClickHouse SQL
+	insertGaugeTableSQL = `INSERT INTO %s_gauge (
+    ResourceAttributes,
+    ResourceSchemaUrl,
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+    ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+    TimeUnix,
+    ValueAsDouble,
+    ValueAsInt,
+    Flags,
+    Exemplars.FilteredAttributes,
+	Exemplars.TimeUnix,
+    Exemplars.ValueAsDouble,
+    Exemplars.ValueAsInt,
+    Exemplars.SpanId,
+    Exemplars.TraceId) VALUES `
+	// language=ClickHouse SQL
+	insertSumTableSQL = `INSERT INTO %s_sum (
+    ResourceAttributes,
+    ResourceSchemaUrl,
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+	ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+    TimeUnix,
+    ValueAsDouble,
+    ValueAsInt,
+    Flags,
+    Exemplars.FilteredAttributes,
+	Exemplars.TimeUnix,
+    Exemplars.ValueAsDouble,
+    Exemplars.ValueAsInt,
+    Exemplars.SpanId,
+    Exemplars.TraceId,
+	AggTemp,
+	IsMonotonic) VALUES `
+	// language=ClickHouse SQL
+	insertHistogramTableSQL = `INSERT INTO %s_histogram (
+	ResourceAttributes,
+    ResourceSchemaUrl,
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+    ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+	StartTimeUnix,
+	TimeUnix,
+	Count,
+	Sum,
+	BucketCounts,
+	ExplicitBounds,
+  	Exemplars.FilteredAttributes,
+	Exemplars.TimeUnix,
+    Exemplars.ValueAsDouble,
+    Exemplars.ValueAsInt,
+    Exemplars.SpanId,
+    Exemplars.TraceId,
+	Flags,
+	Min,
+	Max) VALUES `
+	// language=ClickHouse SQL
+	insertExpHistogramTableSQL = `INSERT INTO %s_exponential_histogram (
+	ResourceAttributes,
+    ResourceSchemaUrl,
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+    ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+	StartTimeUnix,
+	TimeUnix,
+	Count,
+	Sum,                   
+    Scale,
+    ZeroCount,
+	PositiveOffset,
+	PositiveBucketCounts,
+	NegativeOffset,
+	NegativeBucketCounts,
+  	Exemplars.FilteredAttributes,
+	Exemplars.TimeUnix,
+    Exemplars.ValueAsDouble,
+    Exemplars.ValueAsInt,
+    Exemplars.SpanId,
+    Exemplars.TraceId,
+	Flags,
+	Min,
+	Max) VALUES `
+	// language=ClickHouse SQL
+	insertSummaryTableSQL = `INSERT INTO %s_summary (
+	ResourceAttributes,
+    ResourceSchemaUrl,
+    ScopeName,
+    ScopeVersion,
+    ScopeAttributes,
+    ScopeDroppedAttrCount,
+    ScopeSchemaUrl,
+    MetricName,
+    MetricDescription,
+    MetricUnit,
+    Attributes,
+	StartTimeUnix,
+	TimeUnix,
+    Count,
+    Sum,
+    ValueAtQuantiles.Quantile,
+	ValueAtQuantiles.Value,
+    Flags) VALUES `
```

Comment:
might be a good idea to move the SQL to its own go file.

Reply:
done